### PR TITLE
Setup for the PostgreSQL , adding migration files

### PR DESCRIPTION
**Pull request by sarasallamm**
31/7/2024
at 5:30

### CODE REVIEW

**Comment by eslamff** [comment]
@sarasallamm @nancyhany44  PR description should be more informative about what exactly the PR is doing .

File: migrations/20240731135626-fix-migration-replacement.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+};
+
+exports.up = function(db) {
+  // Using promisified readFile
+  const readFileAsync = util.promisify(fs.readFile); // Use util.promisify
+  const filePath = path.join(__dirname, 'sqls', '20240731132502-initialize-up.sql');
+  
+  return readFileAsync(filePath, {encoding: 'utf-8'})
+    .then(data => {
+      console.log('Received data:', data);
```
Comment (eslamff) [inline migrations/20240731135626-fix-migration-replacement.js:28]: @sarasallamm @nancyhany44 
console.log may reveal sensitive data that is not intended to be displayed ( future comment ), remove any console.logs in your code before pushing a PR, the right way to display a log is to collect all of them in one place using a third party app or using the framework itself creating a DB table for them.

**Review by eslamff — CHANGES_REQUESTED** [review]
check the comments added

File: migrations/20240731135626-fix-migration-replacement.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      return db.runSql(data);
+    })
+    .then(() => {
+      const password = process.env.USER_PASSWORD || 's123';
```
Comment (eslamff) [inline migrations/20240731135626-fix-migration-replacement.js:32]: an example.env should be added to the project to provide the team members with the needed environment variables yet without revealing any sensitive info in it for example 
env.example may contain variable like this 
'USER_PASSWORD : " " '

File: migrations/20240731135626-fix-migration-replacement.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      const hashedPassword = bcrypt.hashSync(password, 10);
+      
+      return db.runSql(
+        'INSERT INTO public.users(user_id, name, username, password) VALUES ($1, $2, $3, $4)',
```
Comment (eslamff) [inline migrations/20240731135626-fix-migration-replacement.js:36]: if this is used as a seeding query it should have a process just like migration files there are seeder files, about writing sql in the index file its not the best way for isolating/organizing your code u need to move it to a seeder file.

File: migrations/20240731135626-fix-migration-replacement.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        [1111, 'sarah', 'sarahh', hashedPassword]
+      );
+    });
+};
+
+exports.down = function(db) {
+  const readFileAsync = util.promisify(fs.readFile);
+  const filePath = path.join(__dirname, 'sqls', '20240731132502-initialize-down.sql');
```
Comment (eslamff) [inline migrations/20240731135626-fix-migration-replacement.js:44]: its recommended to use an orm for that part, like TYPEORM will make it much easier

File: src/index.ts [hunk 5]
```diff
@@ -1 +1,32 @@
...
+});
+
+async function fetchData(): Promise<void> {
+    let client: PoolClient | undefined;
+  
+    try {
+      client = await pool.connect();
+      try {
+        const res: QueryResult = await client.query('SELECT * FROM "users"');
+        console.log('Connected to PostgreSQL:', res.rows);
```
Comment (eslamff) [inline src/index.ts:21]: why are we fetching data here ? or its just a test ?

Comment (sarasallamm) [reply src/index.ts:21]: yes it was just to test

File: migrations/20240731135626-fix-migration-replacement.js (hunk 4 above)
Comment (sarasallamm) [reply migrations/20240731135626-fix-migration-replacement.js:44]: I will edit it and use it.

File: migrations/20240731135626-fix-migration-replacement.js (hunk 1 above)
Comment (sarasallamm) [reply migrations/20240731135626-fix-migration-replacement.js:28]: Ohh alright, I will make sure to remove it from any code. Thank you so much

**Comment by eslamff** [comment]
@sarasallamm if you finished the comments in this PR , merge it

**Comment by sarasallamm** [comment]
This was the old one and If I merged this one, there will be alot of conflicts with 2nd PR; however, the 2nd PR is the one that contains typeorm which is found in branch review1 so I wanted to finish the comments in the 2nd PR, and merge it with main.

**Comment by eslamff** [comment]
> This was the old one and If I merged this one, there will be alot of conflicts with 2nd PR; however, the 2nd PR is the one that contains typeorm which is found in branch review1 so I wanted to finish the comments in the 2nd PR, and merge it with main.

PRs should be separated with a single responsibility

**Comment by sarasallamm** [comment]
Alrightt, will focus on that in the upcoming PR.

**Pull request closed without merging**